### ISSUE TITLE
Rule Change: getter-return

### ISSUE THREAD

**Issue by chatquill** [eslint/eslint#20003]
### What rule do you want to change?

getter-return

### What change do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```js
get image() {
    const base64String = this.configNotSaved.getString('backgroundImage');
    if (base64String) return base64ToFile(base64String, 'background-image');
    return;
  }
```

### What does the rule currently do for this code?

error  Expected to return a value in getter 'image'

### What will the rule do after it's changed?

The rule should accept `return;` instead of explicitly requiring `return undefined;`

### Participation

- [ ] I am willing to submit a pull request to implement this change.

### Additional comments

Is there any reason why the rule is raising error on `return;` ?

**Comment by Tanujkanti4441** [comment]
Hi @chatquill, thanks for the issue.

`getter-return` rule has an option called `allowImplicit`, you can enable it to allow `return;`, and let us know if this solves your issue.

[Playground](https://eslint.org/play/#eyJ0ZXh0IjoiLyplc2xpbnQgZ2V0dGVyLXJldHVybjogW1wiZXJyb3JcIiwgeyBhbGxvd0ltcGxpY2l0OiB0cnVlIH1dKi9cbmNvbnN0IHAgPSB7XG4gICAgZ2V0IGltYWdlKCkge1xuICAgIGNvbnN0IGJhc2U2NFN0cmluZyA9IHRoaXMuY29uZmlnTm90U2F2ZWQuZ2V0U3RyaW5nKCdiYWNrZ3JvdW5kSW1hZ2UnKTtcbiAgICBpZiAoYmFzZTY0U3RyaW5nKSByZXR1cm4gYmFzZTY0VG9GaWxlKGJhc2U2NFN0cmluZywgJ2JhY2tncm91bmQtaW1hZ2UnKTtcbiAgICByZXR1cm47XG4gIH1cbn07Iiwib3B0aW9ucyI6eyJydWxlcyI6e30sImxhbmd1YWdlT3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFGZWF0dXJlcyI6e319fX19)

**Comment by chatquill** [comment]
@Tanujkanti4441 Thank you for the prompt answer! Is there a reason why this is not the default behaviour and has to be set using an option?

**Comment by Tanujkanti4441** [comment]
> [@Tanujkanti4441](https://github.com/Tanujkanti4441) Thank you for the prompt answer! Is there a reason why this is not the default behaviour and has to be set using an option?

As per the discussion in https://github.com/eslint/eslint/pull/8460#issuecomment-300054181, the reason is to avoid the conflict with different ESLint rules like `no-useless-return` and `no-undefined` by default.

**Comment by chatquill** [comment]
Thank you!
